db-engine/src/main/java/com/splicemachine/db/iapi/types/ListDataType.java

```diff
@@ -0,0 +1,565 @@
+        return compare(arg, false);
+    }
+    
+    @Override
+    public int compare(DataValueDescriptor other, boolean nullsOrderedLow)
+        throws StandardException {
+        assert other instanceof ListDataType : "illegal comparison of list with non-list.";
+        assert other.getLength() == getLength() : "Trying to compare unequal length lists.";
+        if (this.isNull() || other.isNull()) {
+            if (!isNull())
+                return nullsOrderedLow ? 1 : -1;
+            if (!other.isNull())
+                return nullsOrderedLow ? -1 : 1;
+            return 0; // both null
+        }
+        int result;
+        for (int i = 0; i < numElements; i++) {
+            result = dvd[i].compare(((ListDataType) other).getDVD(i));
+            // List data can be thought of as a concatenation of all the
+            // list data items, so as soon as one part of that "key" is
+            // greater than or less than the other, we know the result.
+            if (result != 0)
+                return result;
+        }
+        return 0;
+    }
+    
+    @Override
+    public boolean compare(int op,
+						   DataValueDescriptor other,
+						   boolean orderedNulls,
+						   boolean unknownRV)
+		throws StandardException
+	{
+        return compare(op, other, orderedNulls, false, unknownRV);
+	}
+ 
+	@Override
+    public boolean compare(int op,
+                           DataValueDescriptor other,
+                           boolean orderedNulls,
+                           boolean nullsOrderedLow,
+                           boolean unknownRV) throws StandardException {
+        
+        assert other != null : "argument is null";
+        assert other instanceof ListDataType : "illegal comparison of list with non-list.";
+        assert other.getLength() == getLength() : "Trying to compare unequal length lists.";
+        
+        // If any of the DVDs in the left or right are null, the
+        // entire ANDed expression is null.
+        if (!orderedNulls && (this.isNull() || other.isNull()))
+            return unknownRV;
+        
+        int result;
+        
+        for (int i = 0; i < numElements; i++) {
+            result = dvd[i].compare(((ListDataType) other).getDVD(i), nullsOrderedLow);
+            // If the "i"th predicate is false, the entire ANDed
+            // expression is false.
+            if (!predSatisfied(result, op))
+                return false;
+        }
+        return true;
+    }
+	/*
+	 * String display of values
+	 */
+
+	public String toString()
+	{
+	    try {
+            return getString();
+        }
+        catch (Exception e) {
+	        return "";
+        }
+	}
+    
+    /*
+     * Hash code
+     */
+    @Override
+    public int hashCode() {
+        final int prime = 37;
+        int result = 17;
+    
+        for (int i = 0; i < numElements; i++) {
+            result = result * prime + dvd[i].hashCode();
```

Comment:
We need to check null before reference dvd[i] here.

Reply:
Added the check.